### ISSUE TITLE
devise a way to store GO-type evidence codes for edges in KG2

### ISSUE THREAD

**Issue by saramsey** [RTXteam/RTX#647]
requested by UAB/PMI team

**Comment by ericawood** [comment]
@saramsey Is there currently a spot in KG2 to store ECO codes/evidence codes/confidence scores/etc? I'm assuming it would go in the publications info dictionary, but I don't know what the key would be? (GO -- #838 -- has ECO codes and evidence codes; CTD -- #695 -- has confidence scores)

**Comment by saramsey** [comment]
Here is Deepak's reply:

![Screen Shot 2020-08-24 at 10 17 06 AM](https://user-images.githubusercontent.com/5562107/91075635-41aea680-e5f3-11ea-8bff-bc5470f15ea0.png)

**Comment by saramsey** [comment]
In KG2, let's store the GO evidence code in an edge slot `evidence`. I think we should store the GO evidence code as as a CURIE ID, like this: `GO:IDA`

**Comment by ericawood** [comment]
> In KG2, let's store the GO evidence code in an edge slot `evidence`. I think we should store the GO evidence code as as a CURIE ID, like this: `GO.EC:IDA`. In `curies-to-urls-map.yaml`, we can map the CURIE prefix `GO.EC` to the base URL `http://www-legacy.geneontology.org/GO.evidence.shtml#`

Hi @saramsey, will both GO evidence codes and ECO evidence codes go in this field? If so, should it be a list? Proposed change to kg2_util:
```
def make_edge(subject_id: str,
              object_id: str,
              relation_curie: str,
              predicate_label: str,
              provided_by: str,
              update_date: str = None):

    return {'subject': subject_id,
            'object': object_id,
            'edge_label': predicate_label,
            'relation': relation_curie,
            'negated': False,
            'publications': [],
            'publications_info': {},
            'update_date': update_date,
            'provided_by': provided_by,
            'evidence': []}
```

`evidence` can be manipulated within each ETL script as necessary and added to an edge via `edge['evidence'] = some_list`.

**Comment by saramsey** [comment]
Yes, let's go with a list of CURIE IDs. Thank you.

**Comment by ericawood** [comment]
Hi @saramsey,
I saw this graphic in the All Things Provenance breakout group:
![image](https://user-images.githubusercontent.com/36611732/94032042-68880600-fd74-11ea-8aa2-249e3228ffbe.png)

Does that mean the entry should be `has_evidence` rather than `evidence`?

**Comment by saramsey** [comment]
> Hi @saramsey,
> I saw this graphic in the All Things Provenance breakout group:
> ![image](https://user-images.githubusercontent.com/36611732/94032042-68880600-fd74-11ea-8aa2-249e3228ffbe.png)
> 
> Does that mean the entry should be `has_evidence` rather than `evidence`?

Good catch. Sure, we can use `has_evidence`.  It may make our life (specifically, export to TSV and import into Neo4j and mediKanren) simpler if we have *every* KG2 edge dict have a `has_evidence` list, which by default will be empty. Does that make sense?

**Comment by ericawood** [comment]
Hi @saramsey,
That makes sense. I will commit that change to a branch so that you can review it, if that works for you.

**Comment by saramsey** [comment]
@ericawood thinks that DrugBank has some evidence code type things that could be put in the `has_evidence` property.

**Comment by saramsey** [comment]
Need to audit the KG2 code base to make sure that in every module where we create an edge, we are either doing so via kg2_util.make_edge or that we add the new edge property `has_evidence` somehow

**Comment by saramsey** [comment]
This issue seems like a good one to bring back to the front-burner.

**Comment by kvarforl** [comment]
UniprotKB has a bunch of evidence codes too; some are associated with the names (which I moved to the description as part of #1171), and others are associated with gene synonyms (isolated but not used anywhere as a part of #1259)

Example of evidence code associated with a gene synonym: `UniProtKB:Q9Y4F9`
```
GN   Name=RIPOR2;
GN   Synonyms=C6orf32, DIFF48, FAM65B, KIAA0386,
GN   PL48 {ECO:0000303|PubMed:9055809};
```

Types of evidence codes are documented [here](https://www.uniprot.org/docs/userman.htm#ev_description)

**Comment by ericawood** [comment]
Sources that do NOT appear to have evidence codes (this will be a long, edited comment as I gather more information):

- repoDB
- UniChem
